src/blocks/ImageBlock.tsx

```diff
@@ -1,111 +1,68 @@
-import Block from './Block'
-import { BlockModel } from './types'
+import React from 'react';
+import Block from './Block';
+import { BlockModel } from './types';
 import BlockFactory from './BlockFactory';
-import './BlockStyles.css'
-import Box from "@mui/material/Box";
-import TextField from "@mui/material/TextField";
-import Button from "@mui/material/Button";
-import Typography from "@mui/material/Typography";
-import UploadFormComponent from './utils/UploadFormComponent';
+import './BlockStyles.css';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import 'swiper/css';
+import ImageUploadPreview from './utils/ImageUploadPreview';
+import ImageWithModal from './utils/ImageWithModal';
 
 export default class ImageBlock extends Block {
-  render() {
-    if (Object.keys(this.model.data).length === 0) {
-      return BlockFactory.renderEmptyState(this.model, this.onEditCallback!)
+  render(): React.ReactNode {
+    let urls = this.model.data['urls'] ? JSON.parse(this.model.data['urls']) : [];
+    if (!urls.length && this.model.data['url']) {
+      urls = [this.model.data['url']]; // Normalize to array format
     }
 
-    let url = this.model.data["url"]
-    if (!url) {
-      return this.renderErrorState()
+    if (urls.length === 0) {
+      return this.renderErrorState();
     }
 
-    return (
-      <img
-        src={url}
-        title="image"
-        style={{
-          height: `100%`,
-          width: `100%`
-        }}
-        onError={(e) => e.currentTarget.src = 'https://www.shutterstock.com/image-illustration/no-picture-available-placeholder-thumbnail-600nw-2179364083.jpg'}
-      />
-    );
-  }
-
-  addHTTPS(url: string) {
-    return (url.indexOf('://') === -1) ? 'http://' + url : url
+    if (urls.length === 1) { // for only one image have it full block width
+      return (
+        <div style={{ display: 'block', width: '100%' }}>
```

Comment:
should retain previous styling for backwards compatability (for image blocks on peoples pages)